File: Programmers/python/네트워크.py
from collections import deque

def set_edge(computers, n):
    network = [[] for _ in range(n)]
    for i in range(n):
        for k in range(i, n):
            if i == k :
                continue
            if computers[i][k] == 1:
                network[i].append(k)
                network[k].append(i)
    
    return network

def bfs(i, isVisited, network):
    dq = deque([i])
    
    while dq:
        cur = dq.popleft()
        for next_com in network[cur]:
            if not isVisited[next_com]:
                dq.append(next_com)
                isVisited[next_com] = True
    
    return isVisited
    

def solution(n, computers):
    answer = 0
    isVisited = [False for _ in range(n)]
    network = set_edge(computers, n)
    
    for i in range(n):
        if not isVisited[i]:
            isVisited = bfs(i, isVisited, network)
            answer += 1
    
    return answer

File: Programmers/java/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        List<Integer>[] graph = new ArrayList[n];
        boolean[] isVisited = new boolean[n];
        
        for(int i = 0; i < n; i++) {
            graph[i] = new ArrayList<>();
        }
        
        for(int i = 0; i < computers.length; i++) {
            for(int k = 0; k < computers[0].length; k++) {
                if(i == k)  continue;
                if(computers[i][k] == 0)    continue;
                graph[i].add(k);
            }
        }
        
        for(int i = 0; i < n; i++) {
            if(!isVisited[i]) {
                dfs(graph, isVisited, i);
                answer++;
            }
        }
        
        return answer;
    }
    
    private void dfs(List<Integer>[] graph, boolean[] isVisited, int cur) {
        isVisited[cur] = true;
        
        List<Integer> nextNodes = graph[cur];
        for(int next : nextNodes) {
            if(isVisited[next]) continue;
            isVisited[next] = true;
            dfs(graph, isVisited, next);
        }
    }
    
    private void bfs(List<Integer>[] graph, boolean[] isVisited, int start) {
        Queue<Integer> queue = new LinkedList<>();
        
        queue.add(start);
        isVisited[start] = true;
        
        while(!queue.isEmpty()) {
            int cur = queue.poll();
            
            List<Integer> nextNodes = graph[cur];
            for(int next : nextNodes) {
                if(isVisited[next]) continue;
                isVisited[next] = true;
                queue.add(next);
            }
        }
    }
}
